pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -408,22 +459,34 @@ def _n_cores_available(self, chip, key, processor_id):
         n_cores = 0
         if processor_id is not None:
             if key in self._core_tracker:
-                if processor_id in self._core_tracker[key]:
-                    return 1
-                return 0
-            elif (key in self._core_tracker and
-                    processor_id in self._core_tracker[key]):
-                return 1
+
+                # check for arbitrary pre allocated cores.
+                if (len(self._core_tracker[key]) -
+                        self._previous_allocated_resources[key] > 0):
+                    if processor_id in self._core_tracker[key]:
+                        return 1
+                    return 0
+                else:
+                    return 0
             elif key not in self._core_tracker:
-                processor = chip.get_processor_with_id(processor_id)
-                if processor is not None and not processor.is_monitor:
-                    return 1
-                return 0
+                # check for arbitrary pre allocated cores.
+                if (chip.n_user_processors -
+                        self._previous_allocated_resources[key] > 0):
+                    # verify the processor is not a monitor core
+                    processor = chip.get_processor_with_id(processor_id)
+                    if processor is not None and not processor.is_monitor:
+                        return 1
+                    return 0
+                else:
+                    return 0
```

Comment:
Would it be possible to reduce the lines of code here a bit by having the `return 0` as a statement that comes after the `if` stack?